arrow-core-data/src/main/kotlin/arrow/core/NonEmptyList.kt

```diff
@@ -282,6 +312,236 @@ class NonEmptyList<out A>(
+      mapN(b, c) { b, c ->
+        Tuple2(b, c)
+      }
+
+    fun <B, C, D> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>
+    ): NonEmptyList<Tuple3<B, C, D>> =
+      mapN(b, c, d) { b, c, d ->
+        Tuple3(b, c, d)
+      }
+
+    fun <B, C, D, E> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>
+    ): NonEmptyList<Tuple4<B, C, D, E>> =
+      mapN(b, c, d, e) { b, c, d, e ->
+        Tuple4(b, c, d, e)
+      }
+
+    fun <B, C, D, E, F> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>,
+      f: NonEmptyList<F>
+    ): NonEmptyList<Tuple5<B, C, D, E, F>> =
+      mapN(b, c, d, e, f) { b, c, d, e, f ->
+        Tuple5(b, c, d, e, f)
+      }
+
+    fun <B, C, D, E, F, G> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>,
+      f: NonEmptyList<F>,
+      g: NonEmptyList<G>
+    ): NonEmptyList<Tuple6<B, C, D, E, F, G>> =
+      mapN(b, c, d, e, f, g) { b, c, d, e, f, g ->
+        Tuple6(b, c, d, e, f, g)
+      }
+
+    fun <B, C, D, E, F, G, H> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>,
+      f: NonEmptyList<F>,
+      g: NonEmptyList<G>,
+      h: NonEmptyList<H>
+    ): NonEmptyList<Tuple7<B, C, D, E, F, G, H>> =
+      mapN(b, c, d, e, f, g, h) { b, c, d, e, f, g, h ->
+        Tuple7(b, c, d, e, f, g, h)
+      }
+
+    fun <B, C, D, E, F, G, H, I> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>,
+      f: NonEmptyList<F>,
+      g: NonEmptyList<G>,
+      h: NonEmptyList<H>,
+      i: NonEmptyList<I>
+    ): NonEmptyList<Tuple8<B, C, D, E, F, G, H, I>> =
+      mapN(b, c, d, e, f, g, h, i) { b, c, d, e, f, g, h, i ->
+        Tuple8(b, c, d, e, f, g, h, i)
+      }
+
+    fun <B, C, D, E, F, G, H, I, J> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>,
+      f: NonEmptyList<F>,
+      g: NonEmptyList<G>,
+      h: NonEmptyList<H>,
+      i: NonEmptyList<I>,
+      j: NonEmptyList<J>
+    ): NonEmptyList<Tuple9<B, C, D, E, F, G, H, I, J>> =
+      mapN(b, c, d, e, f, g, h, i, j) { b, c, d, e, f, g, h, i, j ->
+        Tuple9(b, c, d, e, f, g, h, i, j)
+      }
+
+    fun <B, C, D, E, F, G, H, I, J, K> tupledN(
+      b: NonEmptyList<B>,
+      c: NonEmptyList<C>,
+      d: NonEmptyList<D>,
+      e: NonEmptyList<E>,
+      f: NonEmptyList<F>,
+      g: NonEmptyList<G>,
+      h: NonEmptyList<H>,
+      i: NonEmptyList<I>,
+      j: NonEmptyList<J>,
+      k: NonEmptyList<K>
+    ): NonEmptyList<Tuple10<B, C, D, E, F, G, H, I, J, K>> =
+      mapN(b, c, d, e, f, g, h, i, j, k) { b, c, d, e, f, g, h, i, j, k ->
+        Tuple10(b, c, d, e, f, g, h, i, j, k)
+      }
```

Comment:
```suggestion
```

We discussed to deprecate `tupledN` since it's just an alias for `mapN` with `::TupleN`, and thus doesn't provide a strong benefit to justify so much code / binary. This is more in line with the Kotlin Std.

Reply:
I'm opening a PR asap to remove this from the already finished data types.
We've already deprecated `parTupledN` in Arrow Fx Coroutines as well.